addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/TPLinkSmartHomeBindingConstants.java

```diff
@@ -0,0 +1,123 @@
+
+    // List of all Thing Type UIDs
+    public static final ThingTypeUID THING_TYPE_HS100 = new ThingTypeUID(BINDING_ID, "hs100");
+    public static final ThingTypeUID THING_TYPE_HS105 = new ThingTypeUID(BINDING_ID, "hs105");
+    public static final ThingTypeUID THING_TYPE_HS110 = new ThingTypeUID(BINDING_ID, "hs110");
+    public static final ThingTypeUID THING_TYPE_HS200 = new ThingTypeUID(BINDING_ID, "hs200");
+    public static final ThingTypeUID THING_TYPE_LB100 = new ThingTypeUID(BINDING_ID, "lb100");
+    public static final ThingTypeUID THING_TYPE_LB110 = new ThingTypeUID(BINDING_ID, "lb110");
+    public static final ThingTypeUID THING_TYPE_LB120 = new ThingTypeUID(BINDING_ID, "lb120");
+    public static final ThingTypeUID THING_TYPE_LB130 = new ThingTypeUID(BINDING_ID, "lb130");
+    public static final ThingTypeUID THING_TYPE_RE270K = new ThingTypeUID(BINDING_ID, "re270");
+    public static final ThingTypeUID THING_TYPE_RE370K = new ThingTypeUID(BINDING_ID, "re370");
+
+    /**
+     * All supported Smart Devices.
+     */
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = ImmutableSet.of(THING_TYPE_HS100, THING_TYPE_HS105,
+            THING_TYPE_HS110, THING_TYPE_HS200, THING_TYPE_LB100, THING_TYPE_LB110, THING_TYPE_LB120, THING_TYPE_LB130,
+            THING_TYPE_RE270K, THING_TYPE_RE370K);
+    /**
+     * All Smart Home on/off devices, like switch and plugs.
+     */
+    public static final Set<ThingTypeUID> SWITCH_THING_TYPES = ImmutableSet.of(THING_TYPE_HS100, THING_TYPE_HS105,
+            THING_TYPE_HS200, THING_TYPE_RE270K, THING_TYPE_RE370K);
+    /**
+     * All Smart Home on/off devices that also can monitor energy. Only HS110 supports this.
+     */
+    public static final Set<ThingTypeUID> SWITCH_ENERY_THING_TYPES = ImmutableSet.of(THING_TYPE_HS110);
+
+    public static final Set<ThingTypeUID> BULB_THING_TYPES = ImmutableSet.of(THING_TYPE_LB100, THING_TYPE_LB110,
+            THING_TYPE_LB120, THING_TYPE_LB130);
+
+    // List of default refresh rates in seconds.
+    /**
+     * Default refresh rate for Smart Plugs.
+     */
+    public static final int REFRESH_SMART_PLUG = 30;
+    /**
+     * Default refresh rate for Smart Switches. Must lower because when the devices is switches an action needs to take
+     * place..
+     */
+    public static final int REFRESH_SMART_SWITCH = 1;
+
+    // List of all channel ids
+    public static final String CHANNEL_SWITCH = "switch";
+
+    // List of all plug channel ids
+    public static final String CHANNEL_LED = "led";
+
+    // List of all Light channel ids
+    public static final String CHANNEL_BRIGHTNESS = "brightness";
+    public static final String CHANNEL_COLOR = "color";
+    public static final String CHANNEL_COLOR_TEMPERATURE_KELVIN = "colorTemperatureKelvin";
+    public static final String CHANNEL_COLOR_TEMPERATURE = "colorTemperature";
+    public static final int[] COLOR_TEMPERATURE_LB120_MIN_MAX = { 2700, 6500 };
```

Comment:
AFAIK a `static final int[]` is mutable, maybe using the to separate fields is safer and most likely you will not use them as an array anyway.